### ISSUE TITLE
Make `git go-patch fp` set the author to a bot account

### ISSUE THREAD

**Issue by dagood** [microsoft/go#383]
The `From: You <you@example.org>` line in a patch file can cause problems down the line. I think we could have `fp` automatically change it after formatting the patch files. (To, say, `From: microsoft-golang-bot <microsoft-golang-bot@users.noreply.github.com>`.)

---

I'm thinking about this because of a situation we brushed against in .NET source-build. Over the course of maintenance, a patch file will probably be changed by various people. No one person may know the full history of that patch. Having one specific author called out in the text of the patch file can be misleading: you should look at the Git history of the patch file, instead.

This gets a little worse when the patch file is applied to the repo, because the commit log claims one author wrote it, and the history of the patch file isn't readily available. I'd argue that it's better to hit a wall with the commit author (and hopefully remember that the patch's history is the best place to get this info) rather than go ahead and get in contact with the wrong person.

Another awkward scenario is upstreaming. When a patch file is sent upstream, the From line sets up a supposed author, even though they might not have actually written a substantial amount of the current content. When you have a commit with a bot account author,  instead, it's more obvious that you need to amend the commit with a new author.

---

I think we would need to have `fp` read/write the patch file manually to get this done. `git format-patch` has a promising `--from=<...>` option, but it's just to create a sendable email file: it preserves the commit's original author.

**Comment by gdams** [comment]
@dagood can we close this now?

**Comment by dagood** [comment]
Yep, I can only guess that I reopened because https://github.com/microsoft/go/pull/1787 wasn't done yet, but missed closing it when the PR got merged.
